modules/json/test/src/smithy4s/json/SchemaVisitorJCodecCustomisationTests.scala

```diff
@@ -178,4 +202,39 @@ class SchemaVisitorJCodecCustomisationTests extends FunSuite {
 
     expect(result.f.isNegInfinity)
   }
+
+  test("decoding JSON string as a Float") {
```

Comment:
**question:** something I don't get is: this works because InfinitySupport is on. Why does InfinitySupport have to imply that any other "string number" is valid?

Reply:
that's a sacrifice I'm agreeing to, in order to avoid having to implement N permutations, as the implementations are not compositional. 

I think agreeing to be generally lenient when you want infinity enabled is an okay compromise, considering it applies a similar "let's see if the next character is a quote" piece of logic.